### DISCUSSION TITLE
iam-policy no longer supports requestParameter.userName

### DISCUSSION THREAD

**Discussion by zachzeid** [cloud-custodian/cloud-custodian#7311] [General]
iam-policy in v0.8.20 returned and supported both `requestParameter.username` as well as `requestParameter.policyArn`.  This allowed for auto-remediating functions through `invoke-lambda` to disallow specific IAM policies to be attached, or created.

**Comment by kapilt** [comment]
needs a little more info to investigate. @zachzeid do you have a sample policy that exhibits ?

**Comment by zachzeid** [comment]
@kapilt Was away, here is the gist with the policy and the error I receive.  I am currently testing `AttachUserPolicy`
https://gist.github.com/zachzeid/0c3dcc6d2cd968fe96f4ac5540cdcd02

**Comment by kapilt** [comment]
it seems like the key expression is returning something both username and the policyarn, i'd suggest playing with the key expression so that it just returns the policyarn.
